rosbag2/src/rosbag2/sequential_reader.cpp

```diff
@@ -71,9 +76,29 @@ SequentialReader::open(
   }
 }
 
+std::string SequentialReader::get_next_file()
+{
+  assert(current_file_iterator_ != file_paths_.end());
+  current_file_iterator_++;
+  return *current_file_iterator_;
+}
```

Comment:
I am having a bit of a problem with that implementation, because it does do two things in fact. It increases the file iterator and returns the new current file.
I would recommend either changing the name to something like `void load_next_file()` or let `get_next_file() const` really only return the next path and the iterator has to be advanced outside of the scope of the function.